### ISSUE TITLE
StreamBridge send to bindingName & binderName ignores binding property's content type

### ISSUE THREAD

**Issue by KevinLiLu** [spring-cloud/spring-cloud-stream#2805]
**Describe the issue**
When using StreamBridge's `send(String bindingName, @Nullable String binderName, Object data)` API, it ignores the binding's configured content type and instead forces `application/json`.

https://github.com/spring-cloud/spring-cloud-stream/blob/0c256343c7b063d9ef44b7b96df95a0f8b1f357c/core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/StreamBridge.java#L150

**To Reproduce**
1. Configure binding to use something other than `application/json` content type
2. Call StreamBridge's `send(String bindingName, @Nullable String binderName, Object data)` API

**Version of the framework**
Latest 4.0.4

**Expected behavior**
If content type is not provided in the API, then it should first try to use the binding's configured content type (if binding exists and content type is configured). If binding does not exist or content type is not configured, then it should default to `application/json`. This is the behavior seen in the other send API which does not have content type as a parameter: https://github.com/spring-cloud/spring-cloud-stream/blob/0c256343c7b063d9ef44b7b96df95a0f8b1f357c/core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/StreamBridge.java#L138-L142

**Screenshots**
None

**Additional context**
None

**Comment by KevinLiLu** [comment]
I'm willing to raise a PR if we agree on the expected behavior!

**Comment by KevinLiLu** [comment]
@sobychacko @olegz Thoughts?

**Comment by sobychacko** [comment]
@KevinLiLu We will get back to you soon on this. Thanks!

**Comment by sobychacko** [comment]
@KevinLiLu This was a bug, and thanks for pointing it out. The fix was straightforward and already merged upstream. Please take it for a spin and see if it works for your use case.

**Comment by sobychacko** [comment]
@KevinLiLu My sincere apologies. I saw that earlier, but while investigating the issue, I realized that the fix is so simple that I decided to issue a PR directly from my triaging. We highly encourage you to contribute PRs if you see issues like these. Just ping on the issue and let us know that you are working on it so that we are aware of it. Sorry again for the miscommunication on this matter. Your willingness to help is greatly appreciated and looking forward to your contributions.
